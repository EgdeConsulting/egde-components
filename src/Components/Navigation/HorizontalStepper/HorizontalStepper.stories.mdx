import {
    Meta,
    Story,
    Props,
    Canvas,
    Preview,
} from '@storybook/addon-docs';
import { HorizontalStepper } from './HorizontalStepper';
import { ComponentStability } from 'Icons';
import { LibPath } from 'Types';
import { useState } from 'react';
import { Button } from '@chakra-ui/react';

export const Template = (args) => {
    const [activeStep, setActiveStep] = useState(0);
    return (
        <div>
            <HorizontalStepper
                {...args}
                setActiveStep={setActiveStep}
                activeStep={activeStep}
            />
        </div>
    );
};

<Meta
    title={LibPath.Navigation + 'HorizontalStepper'}
    component={HorizontalStepper}
    args={{
        activeStep: 0,
        setActiveStep: () => {},
        stepsContent: [
            {
                label: 'Step 1',
                children: <div>Content of Step 1</div>,
            },
            {
                label: 'Step 2',
                children: <div>Content of Step 2</div>,
            },
            {
                label: 'Step 3',
                children: <div>Content of Step 3</div>,
            },
            {
                label: 'Step 4',
                children: <div>Content of Step 4</div>,
            },
            {
                label: 'Step 5',
                children: <div>Content of Step 5</div>,
            },
        ],
        buttonTexts: {
            previous: 'previous',
            next: 'next',
            complete: 'complete',
        },
        finalStep: {
            finalText: 'finalText',
            finalButtonText: 'fintalButtonText',
        },
        buttonAlignment: 'left',
        buttonMargin: '0px',
        clickableSteps: false,
        completedSteps: [],
        onFinalize: () => {},
        buttonCaption: {
            defaultValue: {
                text: 'Placeholder',
                color: '#ffffff',
            },
            onFinalStep: () => {},
            isCompleteButtonDisabled: true,
        },
    }}
/>

# Horizontal Stepper

<ComponentStability variant="stable" />

<hr />

This component is used to create a horizontal step-overview which shows the user the step he's on.

<Preview withToolbar>
    <Story name="Base">{Template.bind({})}</Story>
</Preview>

Change some of the props and see how the `Base` component changes.

<Props story="Base" />

# Stories

<hr />

Below are shown different implementations of the component in regards to varying
use cases.

<br />

### Completed steps

<Canvas withToolbar>
    <Story
        name="Completed steps"
        args={{
            activeStep: 4,
            completedSteps: [0, 1, 2, 3, 4],
            isCompleteButtonDisabled: false,
        }}
    >
        {Template.bind({})}
    </Story>
</Canvas>

### Clickable steps

<Canvas withToolbar>
    <Story
        name="Clickable steps"
        args={{
            clickableSteps: true,
            isCompleteButtonDisabled: true,
        }}
    >
        {Template.bind({})}
    </Story>
</Canvas>

### Fewer steps

<Canvas withToolbar>
    <Story
        name="Fewer steps"
        args={{
            stepsContent: [
                {
                    label: 'Step 1',
                    children: <div> Content of Step 1</div>,
                },
                {
                    label: 'Step 2',
                    children: <div>Content of Step 2</div>,
                },
                {
                    label: 'Step 3',
                    children: <div>Content of Step 3</div>,
                },
            ],
            isCompleteButtonDisabled: true,
        }}
    >
        {Template.bind({})}
    </Story>
</Canvas>

### More steps

<Canvas withToolbar>
    <Story
        name="More steps"
        args={{
            stepsContent: [
                {
                    label: 'Step 1',
                    children: <div> Content of Step 1</div>,
                },
                {
                    label: 'Step 2',
                    children: <div>Content of Step 2</div>,
                },
                {
                    label: 'Step 3',
                    children: <div>Content of Step 3</div>,
                },
                {
                    label: 'Step 4',
                    children: <div>Content of Step 4</div>,
                },
                {
                    label: 'Step 5',
                    children: <div>Content of Step 5</div>,
                },
                {
                    label: 'Step 6',
                    children: <div>Content of Step 6</div>,
                },
                {
                    label: 'Step 7',
                    children: <div>Content of Step 7</div>,
                },
                {
                    label: 'Step 8',
                    children: <div>Content of Step 8</div>,
                },
                {
                    label: 'Step 9',
                    children: <div>Content of Step 9</div>,
                },
                {
                    label: 'Step 10',
                    children: <div>Content of Step 10</div>,
                },
                {
                    label: 'Step 11',
                    children: <div>Content of Step 11</div>,
                },
                {
                    label: 'Step 12',
                    children: <div>Content of Step 12</div>,
                },
                {
                    label: 'Step 13',
                    children: <div>Content of Step 13</div>,
                },
                {
                    label: 'Step 14',
                    children: <div>Content of Step 14</div>,
                },
                {
                    label: 'Step 15',
                    children: <div>Content of Step 15</div>,
                },
            ],
            isCompleteButtonDisabled: true,
        }}
    >
        {Template.bind({})}
    </Story>
</Canvas>

### Button Alignment: Left

<Canvas withToolbar>
    <Story
        name="Button Alignment: Left"
        args={{
            buttonAlignment: 'left',
            isCompleteButtonDisabled: true,
        }}
    >
        {Template.bind({})}
    </Story>
</Canvas>

### Button Alignment: Center

<Canvas withToolbar>
    <Story
        name="Button Alignment: Center"
        args={{
            buttonAlignment: 'center',
            isCompleteButtonDisabled: true,
        }}
    >
        {Template.bind({})}
    </Story>
</Canvas>

### Button Alignment: Right

<Canvas withToolbar>
    <Story
        name="Button Alignment: Right"
        args={{
            buttonAlignment: 'right',
            isCompleteButtonDisabled: true,
        }}
    >
        {Template.bind({})}
    </Story>
</Canvas>
