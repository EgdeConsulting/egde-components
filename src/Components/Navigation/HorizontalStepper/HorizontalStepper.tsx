import { Flex, Center, Heading, Button, Text } from '@chakra-ui/react';
import { EditIcon } from 'Icons';
import { Step, Steps } from 'chakra-ui-steps';
import { StepperProps } from 'Types';
import React, { ReactElement } from 'react';

const HorizontalStepper = (props: StepperProps): ReactElement => {
    const {
        activeStep,
        setActiveStep,
        stepsContent,
        buttonTexts,
        finalStep,
        buttonAlignment,
        buttonMargin,
        clickableSteps,
        completedSteps,
        onFinalize,
        buttonCaption,
        onFinalStep,
        isCompleteButtonDisabled,
    } = props;
    const { previous, next, complete } = buttonTexts;
    const { finalText, finalButtonText } = finalStep;

    return (
        <Flex flexDir="column" width="100%">
            <Steps activeStep={activeStep}>
                {stepsContent.map(({ label, children }, index) => (
                    <Step
                        isCompletedStep={
                            completedSteps === undefined
                                ? activeStep > index
                                : completedSteps.includes(index)
                        }
                        _hover={
                            clickableSteps
                                ? {
                                    cursor: 'pointer',
                                    background: 'lightblue',
                                    borderRadius: '20px',
                                }
                                : undefined
                        }
                        label={label}
                        key={`stp${index}`}
                        icon={activeStep === index ? EditIcon : undefined}
                        onClickCapture={
                            clickableSteps ? () => setActiveStep(index) : undefined
                        }
                    >
                        {children}
                    </Step>
                ))}
            </Steps>
            {activeStep === stepsContent.length ? (
                <Center flexDir="column">
                    <Heading>{finalText}</Heading>
                    <Button variant={'primary'} onClick={() => onFinalize()}>
                        {finalButtonText}
                    </Button>
                </Center>
            ) : (
                <Flex
                    width="100%"
                    justify={
                    // eslint-disable-next-line no-nested-ternary
                        buttonAlignment === 'right'
                            ? 'flex-end'
                            : buttonAlignment === 'left'
                                ? 'flex-start'
                                : 'center'
                    }
                    margin={buttonMargin}
                >
                    {activeStep !== 0 && (
                        <Button
                            variant={'secondary'}
                            onClick={() => setActiveStep((prev) => prev - 1)}
                        >
                            {previous}
                        </Button>
                    )}
                    <Flex direction="column">
                        <Button
                            isDisabled={
                                isCompleteButtonDisabled
                        && activeStep === stepsContent.length - 1
                            }
                            variant={'primary'}
                            onClick={
                                activeStep === stepsContent.length - 1
                                    ? () => onFinalStep()
                                    : () => setActiveStep((prev) => prev + 1)
                            }
                        >
                            {activeStep === stepsContent.length - 1 ? complete : next}
                        </Button>
                        {buttonCaption && (
                            <Text color={buttonCaption.color}>{buttonCaption.text}</Text>
                        )}
                    </Flex>
                </Flex>
            )}
        </Flex>
    );
};

export { HorizontalStepper };
