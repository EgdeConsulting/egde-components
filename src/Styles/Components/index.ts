export * from './Text';
export * from './Heading';
export * from './Button';
export * from './IconButton';
export * from './Card';
export * from './Tabs';
export * from './Checkbox';
export * from './Radio';
export * from './Steps';
export * from './Input';
export * from './Textarea';
export * from './Table';
export * from './FileInput';
/* export * from './DatePickerStyles';
 */
