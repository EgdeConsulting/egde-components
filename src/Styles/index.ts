import { extendTheme } from '@chakra-ui/react';
import { colors, fonts } from './Theme';

import {
    Text,
    Heading,
    Button,
    IconButton,
    Card,
    Tabs,
    Checkbox,
    Radio,
    Steps,
    Input,
    Textarea,
    Table,
    FileInput,
} from './Components';

const overrides = {
    colors,
    fonts,
    components: {
        Text,
        Heading,
        Button,
        IconButton,
        Card,
        Tabs,
        Checkbox,
        Radio,
        Steps,
        // DatePicker,
        Input,
        Textarea,
        Table,
        FileInput,
    },
};

const styles = extendTheme(overrides);

export { styles };
