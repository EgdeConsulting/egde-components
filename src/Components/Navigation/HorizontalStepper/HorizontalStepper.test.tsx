import React, { ReactElement, useState } from 'react';
import '@testing-library/jest-dom';
import { render, screen, waitFor  } from '@testing-library/react';
import { HorizontalStepper } from './HorizontalStepper';
import userEvent from '@testing-library/user-event';
import { ChakraProvider } from '@chakra-ui/react';
import { styles } from 'Styles';


const FIRST_STEP_LABEL = 'first step';
const FIRST_STEP_TEXT = 'text of first step';
const SECOND_STEP_LABEL = 'second step';
const SECOND_STEP_TEXT = 'text of second step';
const THIRD_STEP_LABEL = 'third step';
const THIRD_STEP_TEXT = 'text of third step';
const PREVIOUS_BUTTON_TEXT = 'previous';
const NEXT_BUTTON_TEXT = 'next';
const COMPLETE_BUTTON_TEXT = 'complete';
const FINAL_TEXT = 'final text';
const FINAL_BUTTON_TEXT = 'final button text';
const BUTTON_CAPTION_TEXT = 'errors exist';
const BUTTON_ALIGNMENT = 'center';
const BUTTON_MARGIN = '0';

const buttonTexts = {
    previous: PREVIOUS_BUTTON_TEXT,
    next: NEXT_BUTTON_TEXT,
    complete: COMPLETE_BUTTON_TEXT,
};

const finalStep = {
    finalText: FINAL_TEXT,
    finalButtonText: FINAL_BUTTON_TEXT,
};

const testerSteps = [
    {
        label: FIRST_STEP_LABEL,
        children: <>{FIRST_STEP_TEXT}</>,
    },
    {
        label: SECOND_STEP_LABEL,
        children: <>{SECOND_STEP_TEXT}</>,
    },
    {
        label: THIRD_STEP_LABEL,
        children: <>{THIRD_STEP_TEXT}</>,
    },
];

const UnclickableHorizontalStepperTester=(): ReactElement =>{
    const [activeStep, setActiveStep] = useState<number>(0);

    return (
        <ChakraProvider theme={styles}>
            <HorizontalStepper
                activeStep={activeStep}
                setActiveStep={setActiveStep}
                stepsContent={testerSteps}
                buttonTexts={buttonTexts}
                finalStep={finalStep}
                onFinalStep={() => setActiveStep(testerSteps.length)}
                onFinalize={() => setActiveStep(0)}
                buttonAlignment={BUTTON_ALIGNMENT}
                buttonMargin={BUTTON_MARGIN}
            />
        </ChakraProvider>
    );
}
const ClickableHorizontalStepperTester =(): ReactElement =>{
    const [activeStep, setActiveStep] = useState<number>(0);
    const [isPublishClicked, setIsPublishClicked] = useState<boolean>(false);

    return (
        <ChakraProvider theme={styles}>
            <HorizontalStepper
                activeStep={activeStep}
                setActiveStep={setActiveStep}
                stepsContent={testerSteps}
                buttonTexts={buttonTexts}
                finalStep={finalStep}
                clickableSteps
                buttonCaption={
                    activeStep === testerSteps.length - 1 && isPublishClicked
                        ? {
                              text: BUTTON_CAPTION_TEXT,
                              color: 'red',
                          }
                        : undefined
                }
                onFinalStep={() => setIsPublishClicked(true)}
                onFinalize={() => setActiveStep(0)}
                buttonAlignment={BUTTON_ALIGNMENT}
                buttonMargin={BUTTON_MARGIN}
            />
        </ChakraProvider>
    );
}


describe('UnclickableHorizontalStepper component test', () => {
    it('should render correctly', () => {
        const tree = render(<UnclickableHorizontalStepperTester/>);
        expect(tree).toMatchSnapshot();
      });
     
    it('should render UnclickableHorizontalStepper with only next button, step labels and first step content', () => {
        render(<UnclickableHorizontalStepperTester />);
        expect(screen.getByText(FIRST_STEP_LABEL)).toBeVisible();
        expect(screen.getByText(FIRST_STEP_TEXT)).toBeVisible();
        expect(screen.getByText(SECOND_STEP_LABEL)).toBeVisible();
        expect(screen.queryByText(SECOND_STEP_TEXT)).toBeNull();
        expect(screen.getByText(THIRD_STEP_LABEL)).toBeVisible();
        expect(screen.queryByText(THIRD_STEP_TEXT)).toBeNull();
        expect(
            screen.queryByRole('button', {
                name: PREVIOUS_BUTTON_TEXT,
            })
        ).toBeNull();
        expect(
            screen.getByRole('button', {
                name: NEXT_BUTTON_TEXT,
            })
        ).toBeVisible();
    });
    it('should reveal previousButton when nextButton is clicked, and show second step content', async () => {
        render(<UnclickableHorizontalStepperTester />);
        const nextButton = screen.getByRole('button', {
            name: NEXT_BUTTON_TEXT,
        });
        userEvent.click(nextButton);
        await waitFor(() =>
            expect(screen.queryByText(FIRST_STEP_TEXT)).toBeNull()
        );
        expect(await screen.findByText(SECOND_STEP_TEXT)).toBeVisible();
        expect(
            await screen.findByRole('button', {
                name: PREVIOUS_BUTTON_TEXT,
            })
        ).toBeVisible();
    });
    it('should change text to complete on nextButton when clicked two times, and change to final view when clicked the third time', async () => {
        render(<UnclickableHorizontalStepperTester />);
        const nextButton = screen.getByRole('button', {
            name: NEXT_BUTTON_TEXT,
        });
        userEvent.click(nextButton);
        expect(await screen.findByText(SECOND_STEP_TEXT)).toBeVisible();
        userEvent.click(nextButton);
        expect(await screen.findByText(THIRD_STEP_TEXT)).toBeVisible();
        await waitFor(() =>
            expect(nextButton).toHaveTextContent(COMPLETE_BUTTON_TEXT)
        );
        userEvent.click(nextButton);
        expect(await screen.findByText(FINAL_TEXT)).toBeVisible();
        expect(
            await screen.findByRole('button', {
                name: FINAL_BUTTON_TEXT,
            })
        ).toBeVisible();
        await waitFor(() => {
            expect(
                screen.queryByRole('button', {
                    name: PREVIOUS_BUTTON_TEXT,
                })
            ).toBeNull();
            expect(nextButton).not.toBeVisible();
        });
    });
});

describe('ClickableHorizontalStepper component test', () => {
    it('should render correctly', () => {
        const tree = render(<ClickableHorizontalStepperTester/>);
        expect(tree).toMatchSnapshot();
      });

    it('should jump to third step when thirdStepLabelButton is clicked, and show third step content. Should then reveal buttonCaptionText when nextButton is clicked', async () => {
        render(<ClickableHorizontalStepperTester />);
        const nextButton = screen.getByRole('button', {
            name: NEXT_BUTTON_TEXT,
        });
        const thirdStepLabelButton = screen.getByText(THIRD_STEP_LABEL);
        userEvent.click(thirdStepLabelButton);
        await waitFor(() =>
            expect(screen.queryByText(FIRST_STEP_TEXT)).toBeNull()
        );
        expect(await screen.findByText(THIRD_STEP_TEXT)).toBeVisible();
        await waitFor(() =>
            expect(nextButton).toHaveTextContent(COMPLETE_BUTTON_TEXT)
        );
        userEvent.click(nextButton);
        expect(await screen.findByText(BUTTON_CAPTION_TEXT)).toBeVisible();
        expect(
            await screen.findByRole('button', {
                name: PREVIOUS_BUTTON_TEXT,
            })
        ).toBeVisible();
        await waitFor(() => {
            expect(nextButton).toBeVisible();
        });
    });
});
