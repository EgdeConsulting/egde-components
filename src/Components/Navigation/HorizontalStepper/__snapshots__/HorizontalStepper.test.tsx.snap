// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClickableHorizontalStepper component test should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="css-1ofqig9"
      >
        <ol
          class="chakra-steps css-1b8x7s7"
        >
          <li
            aria-disabled="false"
            class="css-1az4mg4"
          >
            <div
              class="css-kcf2wt"
            >
              <div
                aria-current="step"
                class="css-s2gv3l"
              >
                <div
                  class="css-k008qs"
                  style="opacity: 0; transform: scale(0.5) translateZ(0);"
                >
                  <svg
                    class="css-iw1flb"
                    role="img"
                    viewBox="0 0 44 44"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M11.06 6L12 6.94L2.92 16H2V15.08L11.06 6ZM14.66 0C14.41 0 14.15 0.1 13.96 0.29L12.13 2.12L15.88 5.87L17.71 4.04C18.1 3.65 18.1 3 17.71 2.63L15.37 0.29C15.17 0.09 14.92 0 14.66 0ZM11.06 3.19L0 14.25V18H3.75L14.81 6.94L11.06 3.19Z"
                      fill="white"
                    />
                  </svg>
                </div>
              </div>
              <div
                aria-current="step"
                class="css-17ug9r4"
              >
                <span
                  class="css-v0f33p"
                >
                  first step
                </span>
              </div>
            </div>
            <div
              class="css-9m9spt"
            />
          </li>
          <li
            aria-disabled="true"
            class="css-n330yj"
          >
            <div
              class="css-kcf2wt"
            >
              <div
                class="css-s2gv3l"
              >
                <span
                  class="css-eb100c"
                  style="font-weight: medium; text-align: center; opacity: 0; transform: scale(0.5) translateZ(0);"
                >
                  2
                </span>
              </div>
              <div
                class="css-17ug9r4"
              >
                <span
                  class="css-62u96g"
                >
                  second step
                </span>
              </div>
            </div>
            <div
              class="css-9m9spt"
            />
          </li>
          <li
            aria-disabled="true"
            class="css-c6o9co"
          >
            <div
              class="css-kcf2wt"
            >
              <div
                class="css-s2gv3l"
              >
                <span
                  class="css-eb100c"
                  style="font-weight: medium; text-align: center; opacity: 0; transform: scale(0.5) translateZ(0);"
                >
                  3
                </span>
              </div>
              <div
                class="css-17ug9r4"
              >
                <span
                  class="css-62u96g"
                >
                  third step
                </span>
              </div>
            </div>
            <div
              class="css-1i0pmdv"
            />
          </li>
        </ol>
        text of first step
        <div
          class="css-6czo3j"
        >
          <div
            class="css-j7qwjs"
          >
            <button
              class="chakra-button css-1febxxj"
              type="button"
            >
              next
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1ofqig9"
    >
      <ol
        class="chakra-steps css-1b8x7s7"
      >
        <li
          aria-disabled="false"
          class="css-1az4mg4"
        >
          <div
            class="css-kcf2wt"
          >
            <div
              aria-current="step"
              class="css-s2gv3l"
            >
              <div
                class="css-k008qs"
                style="opacity: 0; transform: scale(0.5) translateZ(0);"
              >
                <svg
                  class="css-iw1flb"
                  role="img"
                  viewBox="0 0 44 44"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11.06 6L12 6.94L2.92 16H2V15.08L11.06 6ZM14.66 0C14.41 0 14.15 0.1 13.96 0.29L12.13 2.12L15.88 5.87L17.71 4.04C18.1 3.65 18.1 3 17.71 2.63L15.37 0.29C15.17 0.09 14.92 0 14.66 0ZM11.06 3.19L0 14.25V18H3.75L14.81 6.94L11.06 3.19Z"
                    fill="white"
                  />
                </svg>
              </div>
            </div>
            <div
              aria-current="step"
              class="css-17ug9r4"
            >
              <span
                class="css-v0f33p"
              >
                first step
              </span>
            </div>
          </div>
          <div
            class="css-9m9spt"
          />
        </li>
        <li
          aria-disabled="true"
          class="css-n330yj"
        >
          <div
            class="css-kcf2wt"
          >
            <div
              class="css-s2gv3l"
            >
              <span
                class="css-eb100c"
                style="font-weight: medium; text-align: center; opacity: 0; transform: scale(0.5) translateZ(0);"
              >
                2
              </span>
            </div>
            <div
              class="css-17ug9r4"
            >
              <span
                class="css-62u96g"
              >
                second step
              </span>
            </div>
          </div>
          <div
            class="css-9m9spt"
          />
        </li>
        <li
          aria-disabled="true"
          class="css-c6o9co"
        >
          <div
            class="css-kcf2wt"
          >
            <div
              class="css-s2gv3l"
            >
              <span
                class="css-eb100c"
                style="font-weight: medium; text-align: center; opacity: 0; transform: scale(0.5) translateZ(0);"
              >
                3
              </span>
            </div>
            <div
              class="css-17ug9r4"
            >
              <span
                class="css-62u96g"
              >
                third step
              </span>
            </div>
          </div>
          <div
            class="css-1i0pmdv"
          />
        </li>
      </ol>
      text of first step
      <div
        class="css-6czo3j"
      >
        <div
          class="css-j7qwjs"
        >
          <button
            class="chakra-button css-1febxxj"
            type="button"
          >
            next
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`UnclickableHorizontalStepper component test should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="css-1ofqig9"
      >
        <ol
          class="chakra-steps css-1b8x7s7"
        >
          <li
            aria-disabled="false"
            class="css-1k3dlok"
          >
            <div
              class="css-kcf2wt"
            >
              <div
                aria-current="step"
                class="css-s2gv3l"
              >
                <div
                  class="css-k008qs"
                  style="opacity: 0; transform: scale(0.5) translateZ(0);"
                >
                  <svg
                    class="css-iw1flb"
                    role="img"
                    viewBox="0 0 44 44"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M11.06 6L12 6.94L2.92 16H2V15.08L11.06 6ZM14.66 0C14.41 0 14.15 0.1 13.96 0.29L12.13 2.12L15.88 5.87L17.71 4.04C18.1 3.65 18.1 3 17.71 2.63L15.37 0.29C15.17 0.09 14.92 0 14.66 0ZM11.06 3.19L0 14.25V18H3.75L14.81 6.94L11.06 3.19Z"
                      fill="white"
                    />
                  </svg>
                </div>
              </div>
              <div
                aria-current="step"
                class="css-17ug9r4"
              >
                <span
                  class="css-v0f33p"
                >
                  first step
                </span>
              </div>
            </div>
            <div
              class="css-9m9spt"
            />
          </li>
          <li
            aria-disabled="true"
            class="css-69b456"
          >
            <div
              class="css-kcf2wt"
            >
              <div
                class="css-s2gv3l"
              >
                <span
                  class="css-eb100c"
                  style="font-weight: medium; text-align: center; opacity: 0; transform: scale(0.5) translateZ(0);"
                >
                  2
                </span>
              </div>
              <div
                class="css-17ug9r4"
              >
                <span
                  class="css-62u96g"
                >
                  second step
                </span>
              </div>
            </div>
            <div
              class="css-9m9spt"
            />
          </li>
          <li
            aria-disabled="true"
            class="css-10a7ufa"
          >
            <div
              class="css-kcf2wt"
            >
              <div
                class="css-s2gv3l"
              >
                <span
                  class="css-eb100c"
                  style="font-weight: medium; text-align: center; opacity: 0; transform: scale(0.5) translateZ(0);"
                >
                  3
                </span>
              </div>
              <div
                class="css-17ug9r4"
              >
                <span
                  class="css-62u96g"
                >
                  third step
                </span>
              </div>
            </div>
            <div
              class="css-1i0pmdv"
            />
          </li>
        </ol>
        text of first step
        <div
          class="css-6czo3j"
        >
          <div
            class="css-j7qwjs"
          >
            <button
              class="chakra-button css-1febxxj"
              type="button"
            >
              next
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1ofqig9"
    >
      <ol
        class="chakra-steps css-1b8x7s7"
      >
        <li
          aria-disabled="false"
          class="css-1k3dlok"
        >
          <div
            class="css-kcf2wt"
          >
            <div
              aria-current="step"
              class="css-s2gv3l"
            >
              <div
                class="css-k008qs"
                style="opacity: 0; transform: scale(0.5) translateZ(0);"
              >
                <svg
                  class="css-iw1flb"
                  role="img"
                  viewBox="0 0 44 44"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11.06 6L12 6.94L2.92 16H2V15.08L11.06 6ZM14.66 0C14.41 0 14.15 0.1 13.96 0.29L12.13 2.12L15.88 5.87L17.71 4.04C18.1 3.65 18.1 3 17.71 2.63L15.37 0.29C15.17 0.09 14.92 0 14.66 0ZM11.06 3.19L0 14.25V18H3.75L14.81 6.94L11.06 3.19Z"
                    fill="white"
                  />
                </svg>
              </div>
            </div>
            <div
              aria-current="step"
              class="css-17ug9r4"
            >
              <span
                class="css-v0f33p"
              >
                first step
              </span>
            </div>
          </div>
          <div
            class="css-9m9spt"
          />
        </li>
        <li
          aria-disabled="true"
          class="css-69b456"
        >
          <div
            class="css-kcf2wt"
          >
            <div
              class="css-s2gv3l"
            >
              <span
                class="css-eb100c"
                style="font-weight: medium; text-align: center; opacity: 0; transform: scale(0.5) translateZ(0);"
              >
                2
              </span>
            </div>
            <div
              class="css-17ug9r4"
            >
              <span
                class="css-62u96g"
              >
                second step
              </span>
            </div>
          </div>
          <div
            class="css-9m9spt"
          />
        </li>
        <li
          aria-disabled="true"
          class="css-10a7ufa"
        >
          <div
            class="css-kcf2wt"
          >
            <div
              class="css-s2gv3l"
            >
              <span
                class="css-eb100c"
                style="font-weight: medium; text-align: center; opacity: 0; transform: scale(0.5) translateZ(0);"
              >
                3
              </span>
            </div>
            <div
              class="css-17ug9r4"
            >
              <span
                class="css-62u96g"
              >
                third step
              </span>
            </div>
          </div>
          <div
            class="css-1i0pmdv"
          />
        </li>
      </ol>
      text of first step
      <div
        class="css-6czo3j"
      >
        <div
          class="css-j7qwjs"
        >
          <button
            class="chakra-button css-1febxxj"
            type="button"
          >
            next
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
